spec/integration/admin_spec.rb

```diff
@@ -147,4 +147,30 @@
       end
     end
   end
+
+  describe 'generating an upload url' do
+
+    it 'shows a flash message with instructions on forwarding the url' do
+      request = FactoryGirl.create(:info_request)
+      authority_name = request.public_body.name
+      authority_email = request.public_body.request_email
+
+      using_session(@admin) do
+        visit admin_request_path :id => request.id
+        find_button('Generate URL').click
+
+        url = confirm_url(PostRedirect.last.email_token)
+
+        message = "Send \"#{authority_name}\" &lt;" \
```

Comment:
Since this is an integration spec, I'd definitely use the capybara matchers here.